# Assignment 1: Statistical Interpretations

Now that you have some stats tools under your belts, let's test and see how much you know! This assignment is meant to three things:

1. Can you run some descriptive statistics?
2. Can you pick the right test to answer the question of interest?
3. Can you interpret what the result of your test means?

This assignment will consist of both coding questions and multiple choice questions as well. Topics in the assignment may be directly from lectures/practicals or might encourage you to look for new functions to accomplish the goal. Please ensure to read the questions carefully and to copy variable names and file paths to ensure no typos. As the assignment will be using autograder, it will mark typos as incorrect, and no part marks will be rewarded ... you have been warned. Make sure you also load all the packages required for the functions you are using. You will periodically see blank cells, do not move them or try to delete them as these are the grading cells. As always if you have any questions feel free to post on the discussion board or email both Vicki \([vickimengyuan.zhang@humber.ca](mailto:vickimengyuan.zhang@humber.ca)\) and Yeshoda \([yeshoda.harry\-paul@humber.ca](mailto:yeshoda.harry-paul@humber.ca)\). Good luck!


---

## Section 1: Descriptive Statistics

As mentioned in class, descriptive statistics are a good way to assess your data and find patterns of interest that may be worth investigating further. Here we are going to play with some descriptive statistics.

### Question 1 (3 pts total)

Use the `trees` dataset for the next few questions.

#### Question 1a (1 pts)

Which of the variables in the dataset has the largest spread of data? Store the answer in the variable `question_1a` as a string, for example: `question_1a <- "Answer"`.

#### Question 1b (1 pts)

Which of the variables in the dataset has the smallest difference between the median and the mean?

In [3]:
# Identify the variable with the largest spread
spread_values <- sapply(trees, sd) #The sapply() function is used to apply the sd() (standard deviation) function to each column of the 'trees' dataset.
question_1a <- names(spread_values[which.max(spread_values)])

#Identify the variable with the smallest difference between median and mean
difference_median_mean <- sapply(trees, function(x) abs(median(x) - mean(x))) #sapply() is used to apply a function to each column of the 'trees' dataset. In this case, the function calculates the absolute difference between the median and mean for each column and stores the results in the median_mean_diff vector.
question_1b <- names(difference_median_mean[which.min(difference_median_mean)])

# Print the answers
cat("Answer for Question 1a:", question_1a, "\n")
cat("Answer for Question 1b:", question_1b, "\n")

Answer for Question 1a: Volume 


Answer for Question 1b: Height 


#### Question 1c \(1 pt\)

Continue using the `trees` dataset.

```{kernel = "ir"}
head(trees)
```

Determine which of the two variables has the highest significant correlation -- what is the value of their correlation? Round your answer to two decimal places, and then store your answer in `question_1c` as a number.


In [4]:
# Calculate the correlation matrix for all pairs of variables
cor_matrix <- cor(trees)

# Find the row and column indices of the pair with the highest correlation
max_corr_index <- which(cor_matrix == max(cor_matrix[upper.tri(cor_matrix, diag = FALSE)], na.rm = TRUE), arr.ind = TRUE)

# Get the names of the variables with the highest correlation
variable_names <- colnames(cor_matrix)[max_corr_index[, "col"]]

# Store the answer in the variable question_1c
question_1c <- list(names = variable_names, correlation = round(max(cor_matrix[upper.tri(cor_matrix, diag = FALSE)], na.rm = TRUE), 2))

# Print the result
cat("Answer for Question 1c: The value of", variable_names[1], "and", variable_names[2], "correlation (rounded to two decimal places) is:", question_1c$correlation)

Answer for Question 1c: The value of Girth and Volume correlation (rounded to two decimal places) is: 0.97

In [5]:
# Perform the correlation test between Girth and Volume
cor_test_girth_volume <- cor.test(x = trees$Girth, y = trees$Volume, method = "pearson")

# Print the correlation test results
print(cor_test_girth_volume) 



	Pearson's product-moment correlation

data:  trees$Girth and trees$Volume
t = 20.478, df = 29, p-value < 2.2e-16
alternative hypothesis: true correlation is not equal to 0
95 percent confidence interval:
 0.9322519 0.9841887
sample estimates:
      cor 
0.9671194 



#### Question 1c \(1 pt\)

Continue using the `trees` dataset.

```{kernel = "ir"}
head(trees)
```

Determine which of the two variables has the highest significant correlation -- what is the value of their correlation? Round your answer to two decimal places, and then store your answer in `question_1c` as a number.



A p-value less than alpha (0.05),where p < 0.05, and r = 0.967 (correlation coefficient) means that Girth and Volume has the highest significant correlation.


---

## Section 2: Deciding Tests

In this section, you will have to decide which test is best to use to answer the question provided. Don't forget to check if the assumptions are met!

### Question 2 (3 pts total)

You will be using the data from the `OrchardSprays` to answer the following questions.

```{kernel = "ir"}
head(OrchardSprays)
```

#### Question 2 Part 1 (1.5 pts)

Given the data you've imported, what test would be best for testing each of the following below? Make sure to investigate the data frame, and then save your answer as an object, for example:
`question2a <- "linear model"`

- \(A\) Is the data for each treatment normally distributed? Store your answer as a string in the variable `question2_1a`
- \(B\) What is the effect of the treatment on the decrease in the solution of the volume? Store your answer as a string in the variable `question2_1b`
- \(C\) For only the cells on row 1, what is the difference in the decrease in the solution between treatment A and treatment H? Store your answer as a string in the variable `question2_1c`. Hint: consider whether each row or column have independent cells of solution or not.



#### Question 1c \(1 pt\)

Continue using the `trees` dataset.

```{kernel = "ir"}
head(trees)
```

Determine which of the two variables has the highest significant correlation -- what is the value of their correlation? Round your answer to two decimal places, and then store your answer in `question_1c` as a number.


In [6]:
# Shapiro-Wilk test for normality for each treatment group
shapiro_by_treatment <- tapply(OrchardSprays$decrease, OrchardSprays$treatment, shapiro.test) #tapply is used to apply the shapiro.test function to the "decrease" variable, grouped by the "treatment" variable.

# Check if all p-values are above 0.05
all_normal <- all(sapply(shapiro_by_treatment, function(x) x$p.value > 0.05)) #sapply applies a function to each element of the shapiro_by_treatment list, extracting the p-values from the test results.

# Store the answer in the variable question_2_1a
question2_1a <- ifelse(all_normal, "The data for each treatment is normally distributed", "The data for each treatment is not normally distributed")

#print the results
cat("Answer for Question_2 Part_1 A:", question2_1a, "\n")

Answer for Question_2 Part_1 A: The data for each treatment is not normally distributed 


The Shapiro-Wilk normality test was applied to eight distinct treatment groups (A to H) within the dataset. For groups B, D, F, and G, the obtained p-values exceeded the commonly used significance level of 0.05. This indicates that there is insufficient evidence to reject the null hypothesis, suggesting that the data in these groups is not significantly different from a normal distribution. On the contrary, groups A, C, E, and H exhibited p-values less than or equal to 0.05, providing evidence to reject the null hypothesis. Therefore, it can be inferred that the data in these groups deviates significantly from a normal distribution.

In [7]:
# Perfoem Kruskal-Wallis test for non-normally distributed data
kruskal_result <- kruskal.test(decrease ~ treatment, data = OrchardSprays)

# Check if the p-value is below 0.05
non_normal_test_result <- ifelse(kruskal_result$p.value < 0.05, "There is a significant effect of treatment on the decrease in the solution of the volume", "There is no significant effect of treatment on the decrease in the solution of the volume")

# Store the answer in the variable question_2_1b
question2_1b <- paste("Kruskal-Wallis:", non_normal_test_result, "(H = 48.874, p = 2.402e-08)")

# Print the results
cat("Answer for Question 2 Part_1 B:", question2_1b, "\n")
                         

Answer for Question 2 Part_1 B: Kruskal-Wallis: There is a significant effect of treatment on the decrease in the solution of the volume (H = 48.874, p = 2.402e-08) 


In [8]:
# Subset the data for only row 1
row1_data <- subset(OrchardSprays, rowpos == 1)

# Extract the decrease values for treatment A and treatment H in row 1
decrease_A <- row1_data$decrease[row1_data$treatment == "A"]
decrease_H <- row1_data$decrease[row1_data$treatment == "H"]

# Calculate the difference
difference_decrease_AH <- decrease_A - decrease_H

# Store the answer in the variable question_2_1c
question2_1c <- paste("The difference in the decrease in solution between treatment A and treatment H in row 1 is:", difference_decrease_AH)

# Print the results
cat("Answer for Part_1 C:", question2_1c, "\n")

Answer for Part_1 C: The difference in the decrease in solution between treatment A and treatment H in row 1 is: -118 


#### Question 2 Part 2 (1.5 pts)

You consider the question in (B) the most interesting: What is the effect of the treatment on the decrease in the solution of the volume? Run the test appropriate test and save it into `question2_2`. If you need to run a post-hoc test, save it into `question2_3`.



In [9]:
# Perform Kruskal-Wallis test for non-normally distributed data
kruskal_result <- kruskal.test(decrease ~ treatment, data = OrchardSprays)

# Print the result
print(kruskal_result)

# Check if the p-value is below 0.05
non_normal_test_result <- ifelse(kruskal_result$p.value < 0.05, "significant", "not significant")

# Store the answer in the variable question_2_2
question2_2 <- paste("Kruskal-Wallis test result for the effect of treatment on decrease is", non_normal_test_result)

# Print the results
cat("Answer for Part 2_2:", question2_2, "\n")


	Kruskal-Wallis rank sum test

data:  decrease by treatment
Kruskal-Wallis chi-squared = 48.874, df = 7, p-value = 2.402e-08



Answer for Part 2_2: Kruskal-Wallis test result for the effect of treatment on decrease is significant 


Conducted a Kruskal\-Wallis rank sum test to examine the effect of treatment on the decrease in solution volume. From the output, we can see that there is a significant difference between at least one treatment group \(H = 48.874, p = 2.402e\-08\), but which one? We need to use the post\-hoc test called the Dunn test to test pairwise differences while controlling for the multiple comparisons.


In [10]:
# Perform pairwise Wilcoxon test for post-hoc analysis
suppressWarnings({pairwise_result <- pairwise.wilcox.test(OrchardSprays$decrease, OrchardSprays$treatment, p.adjust.method = "bonferroni")})

# Store the answer in the variable question_2_3
question2_3 <- pairwise_result

# Print the result
cat("Answer for Part 2_3:\n")
print(question2_3)

Answer for Part 2_3:



	Pairwise comparisons using Wilcoxon rank sum test with continuity correction 

data:  OrchardSprays$decrease and OrchardSprays$treatment 

  A     B     C     D     E     F     G    
B 1.000 -     -     -     -     -     -    
C 0.037 0.076 -     -     -     -     -    
D 0.026 0.026 1.000 -     -     -     -    
E 0.026 0.026 0.131 0.503 -     -     -    
F 0.026 0.026 0.131 0.665 1.000 -     -    
G 0.025 0.026 0.379 0.206 1.000 1.000 -    
H 0.025 0.026 0.150 0.026 1.000 1.000 1.000

P value adjustment method: bonferroni 


Wanted to perform a post\-hoc Dunn's test using dunn.test but was getting some errors in installation so pairwise comparisons were performed using the Wilcoxon rank sum test with continuity correction to assess differences in the decrease variable among treatments \(A to H\) in the OrchardSprays dataset. The resulting matrix displays adjusted p\-values, where values below the significance level indicate significant differences. The Bonferroni correction was applied, revealing significant differences between treatments A and C, as well as treatments D and H. This post\-hoc analysis provides detailed insights into specific treatment pairs contributing to the overall observed effect in the Kruskal\-Wallis test.


#### Question 1c \(1 pt\)

Continue using the `trees` dataset.

```{kernel = "ir"}
head(trees)
```

Determine which of the two variables has the highest significant correlation -- what is the value of their correlation? Round your answer to two decimal places, and then store your answer in `question_1c` as a number.


---

### Question 3 (2 pts total)

Consider the `toothgrowth` dataset below. What would be the best design formula to test the effect of the delivery methods (`supp`) on toothgrowth? What about the effect of orange juice (`OJ`) at 0.5mg/day? Assume you're using a `DESeq2` style design formula. Store your answer as a string in `question_3a` and `question_3b`respectively. 

```r {kernel="ir"}
data("ToothGrowth")
head(ToothGrowth)
```



In [4]:
# Design formula for testing the effect of delivery methods (supp) on toothgrowth
question_3a <- ~ supp

# Design formula for testing the effect of orange juice (OJ) at 0.5mg/day
question_3b <- ~ supp + dose + supp:dose

# Print the results
cat("3a: Design formula for the effect of delivery methods (supp) on toothgrowth:", as.character(question_3a), "\n")
cat("3b: Design formula for the effect of orange juice (OJ) at 0.5mg/day:", as.character(question_3b), "\n")

3a: Design formula for the effect of delivery methods (supp) on toothgrowth: ~ supp 


3b: Design formula for the effect of orange juice (OJ) at 0.5mg/day: ~ supp + dose + supp:dose 


---

## Section 3: Interpretation

#### \(2 pts total\)

In this section, you will be given a snippet from a paper and it will be your job to determine:

1. What test did they conduct? Store your answers in `question_4a`.
2. Were their results significant or not? Store your answer as "Y" for yes, or "N" for no in `question_4b`.
3. What do the results mean for their study? Interpret.

> _Chlamydomonas_ growth rate plateaued around 35°C but was largely reduced at 40°C. Chlamydomonas cells \(CC\-1690, 21gr, wildtype\) were grown in photobioreactors \(PBRs\) in Tris\-acetate\-phosphate \(TAP\) medium under turbidostatic conditions at different temperatures with a light intensity of 100 µmol photons m−2 s−1 and constantly bubbling of air. Relative growth rates were calculated based on the cycling of OD680, see Supplementary Fig. 1 and methods for details. OD680 is proportional to total chlorophyll content in unit of µg chlorophyll mL−1. Each temperature treatment was conducted in an individual PBR. Mean ± SE, n = 3 biological replicates. Statistical analyses were performed using two\-tailed t\-test assuming unequal variance by comparing treated samples with pre\-heat or 30°C with 35°C \(\*, p &lt; 0.05, the colors of asterisks match the treatment conditions\). Not significant, ns.
>
> <img src=".Statistical_Interpretations-GOOD.ipynb.upload/Screenshot 2024-01-23 at 11.20.09 PM.png"   width="507.219px"  height="300.602px"  style="object-fit:cover"/>

<span style='color:#d9e3f0'>Taken from: https://www.nature.com/articles/s42003\-022\-03359\-z</span>

In [1]:
# Question 4a: What test did they conduct?
question_4a <- "Welch's test"

# Question 4b: Were their results significant or not?
question_4b <- "Y"  # Yes, their results were significant (indicated by * in the text)

# Print the results
cat("Question 4a: What test did they conduct?\n")
cat(question_4a, "\n\n")

cat("Question 4b: Were their results significant or not?\n")
cat("Results were significant: ", ifelse(question_4b == "Y", "Yes", "No"), "\n")

Question 4a: What test did they conduct?


Welch's test 



Question 4b: Were their results significant or not?


Results were significant:  Yes 


What do the results mean for their study?

The results suggests that there are significant differences in Chlamydomonas growth rates at different temperatures. Specifically, Chlamydomonas growth rate plateaued around 35°C but was largely reduced at 40°C. The use of  Welch's test  indicates a comparison between treated samples with pre\-heat \(30°C\) and 35°C. The statistical significance suggests that the observed differences in growth rates are unlikely due to random chance alone. It implies that temperature conditions significantly influence the growth rates of Chlamydomonas cells. 
